### PR TITLE
Change tests

### PR DESCRIPTION
**Pull request by felipe-colussi**
## Problem
 Tests were printing point address instead of json value.  
 
## Solution

Change the test functions that used  `!reflect.DeepEqual`to  use testify `assert.Equal` 

Example of testify output: 

```

=== RUN   TestFlattenClusterDriver
    structure_cluster_driver_test.go:61: 
        	Error Trace:	/home/user/go/src/terraform-provider-rancher2/rancher2/structure_cluster_driver_test.go:61
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"active":true, "actual_url":"actual_urll", "builtin":true, "checksum":"XXXXXXXX", "name":"name", "ui_url":"ui_url", "url":"url", "whitelist_domains":[]interface {}{"domain1", "domain2"}}
        	            	actual  : map[string]interface {}{"active":true, "actual_url":"actual_url", "builtin":true, "checksum":"XXXXXXXX", "name":"name", "ui_url":"ui_url", "url":"url", "whitelist_domains":[]interface {}{"domain1", "domain2"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,3 +2,3 @@
        	            	  (string) (len=6) "active": (bool) true,
        	            	- (string) (len=10) "actual_url": (string) (len=11) "actual_urll",
        	            	+ (string) (len=10) "actual_url": (string) (len=10) "actual_url",
        	            	  (string) (len=7) "builtin": (bool) true,
        	Test:       	TestFlattenClusterDriver
        	Messages:   	Unexpected output from flattener.
--- FAIL: TestFlattenClusterDriver (0.00s)


Expected :map[string]interface {}{"active":true, "actual_url":"actual_urll", "builtin":true, "checksum":"XXXXXXXX", "name":"name", "ui_url":"ui_url", "url":"url", "whitelist_domains":[]interface {}{"domain1", "domain2"}}
Actual   :map[string]interface {}{"active":true, "actual_url":"actual_url", "builtin":true, "checksum":"XXXXXXXX", "name":"name", "ui_url":"ui_url", "url":"url", "whitelist_domains":[]interface {}{"domain1", "domain2"}}
<Click to see difference>


```

### Manual Testing

 Run it locally to ensure that it didn't change the tests behavior.
 
 Force an error and test the output. Then change it back to normal.

### CODE REVIEW

File: rancher2/structure_auth_config_openldap_test.go
Comment (a-blender) [inline rancher2/structure_auth_config_openldap_test.go]: Nit: extra line/s

File: rancher2/structure_cluster_rke_config_cloud_provider_openstack_test.go
Comment (a-blender) [inline rancher2/structure_cluster_rke_config_cloud_provider_openstack_test.go]: Since we using the testify/assert pkg let's also update these calls to `tf.Fatalf` to an assert call on test failure. Something like https://pkg.go.dev/github.com/stretchr/testify/assert#Assertions.FailNow with the same error msgs. Re-test to make sure behavior hasn't changed

**Review by a-blender — CHANGES_REQUESTED** [review]

File: rancher2/structure_cluster_rke_config_cloud_provider_openstack_test.go
Comment (felipe-colussi) [reply rancher2/structure_cluster_rke_config_cloud_provider_openstack_test.go]: Done

**Review by a-blender — APPROVED** [review]
@felipe-colussi You need to rebase with main but otherwise lgtm as long as the TF tests pass and local `make validate` passes as well. Thanks for working on this

**Review by thatmidwesterncoder — APPROVED** [review]

**Review by jakefhyde — APPROVED** [review]

**Review by jakefhyde — APPROVED** [review]

**Pull request merged**
